### PR TITLE
Added generic dev server 

### PR DESCRIPTION
**Pull request by scott-phillips-sp**
This PR is based upon the feedback on https://github.com/meltedspark/angular-cli-builders/pull/6. It creates a new generic dev server where if the build target is a browserBuilder then it uses that builder's public buildWebpackConfig() method to generate the webpack config.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,7 +56,8 @@
     "typescript": "^2.9.2"
   },
   "peerDependencies": {
-    "@angular-devkit/architect": ">=0.7.0-rc.0",
-    "@angular-devkit/build-angular": ">=0.7.0-rc.0"
+    "@angular-devkit/architect": ">=0.7.0-rc.3",
```
Comment (meltedspark) [inline package.json]: It's preferable to have the least possible version as a dependency. Are you confident `rc.3` is the minimal required version?

File: src/generic/dev-server/index.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+import {BuilderContext, BuilderConfiguration, BuildEvent, Builder} from '@angular-devkit/architect';
+import {DevServerBuilder, DevServerBuilderOptions, BrowserBuilderSchema, BrowserBuilder, NormalizedBrowserBuilderSchema } from '@angular-devkit/build-angular';
+import {Path, virtualFs} from '@angular-devkit/core';
+import {Observable} from 'rxjs';
+import {tap, switchMap} from 'rxjs/operators';
+import * as fs from 'fs';
+
+
+const webpackMerge = require('webpack-merge');
```
Comment (meltedspark) [inline src/generic/dev-server/index.ts]: Any use of this?

File: src/generic/dev-server/index.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+
+import {BuilderContext, BuilderConfiguration, BuildEvent, Builder} from '@angular-devkit/architect';
+import {DevServerBuilder, DevServerBuilderOptions, BrowserBuilderSchema, BrowserBuilder, NormalizedBrowserBuilderSchema } from '@angular-devkit/build-angular';
+import {Path, virtualFs} from '@angular-devkit/core';
+import {Observable} from 'rxjs';
+import {tap, switchMap} from 'rxjs/operators';
+import * as fs from 'fs';
```
Comment (meltedspark) [inline src/generic/dev-server/index.ts]: Please remove unused imports

File: src/generic/dev-server/index.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  buildWebpackConfig(
+    root: Path,
+    projectRoot: Path,
+    host: virtualFs.Host<fs.Stats>,
+    browserOptions: BrowserBuilderSchema,
+  ) {
+    // If our target builder has a webpack config method lets use it. Otherwise we
+    // will fall to the default config
+    if (this.targetBuilder instanceof BrowserBuilder) {
```
Comment (meltedspark) [inline src/generic/dev-server/index.ts]: Actually I would check that the `targetBuilder` has a method `buildWebpackConfig`. It won't be necessarily instance of `BrowserBuilder`. If you go with this assumption it means everyone that wants to implement custom webpack builder will have to inherit from `BrowserBuilder`

File: src/generic/dev-server/schema.d.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { BrowserBuilderSchema } from '@angular-devkit/build-angular';
+
+export interface GenericDevServerBuildSchema extends BrowserBuilderSchema { }
```
Comment (meltedspark) [inline src/generic/dev-server/schema.d.ts]: This seems to be redundant

**Comment by meltedspark** [comment]
Closes #5

File: package.json (hunk 1 above)
Comment (scott-phillips-sp) [reply package.json]: Yes. It needs this commit, which was introduced with rc.3:

https://github.com/angular/angular-cli/commit/cb588b68b34641cbfc31b563f741f94f48211b2d#diff-6c0b312b8bd33db747e50549a66f7ab1

File: src/generic/dev-server/index.ts (hunk 4 above)
Comment (scott-phillips-sp) [reply src/generic/dev-server/index.ts]: 👍

**Comment by mwanago** [comment]
This is just what I've needed. Great! Can't wait for it.

File: src/generic/dev-server/index.ts (hunk 3 above)
Comment (scott-phillips-sp) [reply src/generic/dev-server/index.ts]: fs.stat is used as part of the method signature for the buildWebpackConfig method.... Is there a better way to to import just the stat method from the fs module?

File: src/generic/dev-server/index.ts (hunk 2 above)
Comment (scott-phillips-sp) [reply src/generic/dev-server/index.ts]: 👍

File: src/generic/dev-server/index.ts (hunk 3 above)
Comment (meltedspark) [reply src/generic/dev-server/index.ts]: Yep, like this:
`import {Stats} from 'fs';` and then in sig: ` host: virtualFs.Host<Stats>,`

**Comment by scott-phillips-sp** [comment]
@meltedspark Updates requested have been made. And yes I've tested this with a cli app.

**Pull request merged**